core/src/main/java/org/jdbi/v3/core/generic/GenericTypes.java

```diff
@@ -116,9 +113,20 @@ private GenericTypes() {
      * @return the resolved type.
      */
     public static Type resolveType(Type type, Type contextType) {
-        return TypeToken.of(contextType)
-                .resolveType(type)
-                .getType();
+        return GenericTypeReflector.resolveType(GenericTypeReflector.annotate(type), GenericTypeReflector.annotate(contextType)).getType();
```

Comment:
Please use `GenericTypeReflector.resolveType(Type, Type)` instead of wrapping first using `annotate`. The end result is the same, but it avoids one extra traversal through the type structure, so it should be somewhat faster.

Details, in case you care:
When calling `GenericTypeReflector.resolveType(AnnotatedType, AnnotatedType)`, the `contextType` first gets its generics expanded (so that each `Class` with type parameters gets wrapped into a `ParameterizedType`), and when calling `GenericTypeReflector.resolveType(Type, Type)` this is done at the same time the `Type` is wrapped into an `AnnotatedType`, so one traversal is skipped.

Reply:
Thanks!